src/error.rs

```diff
@@ -169,14 +171,10 @@ impl From<SystemTimeError> for Error {
   }
 }
 
-impl From<walkdir::Error> for Error {
-  fn from(walkdir_error: walkdir::Error) -> Self {
-    let path = walkdir_error.path().unwrap().to_owned();
-
-    if let Some(source) = walkdir_error.into_io_error() {
-      Self::Filesystem { source, path }
-    } else {
-      Self::internal("Encountered walkdir error without source")
+impl From<ignore::Error> for Error {
+  fn from(ignore_error: ignore::Error) -> Self {
+    Self::FileParse {
+      source: ignore_error,
```

Comment:
This feels like a cop out, but with the `ignore` crate implementing it's own enum of different errors, I wasn't sure how to represent it properly without just deferring to that crate's `Display` implementation.